scripts/Common.ps1

```diff
@@ -115,6 +115,73 @@ function ExecuteAndExitOnFailure([string] $command, [bool] $captureOutput = $fal
     return $output
 }
 
+function GetRepoInfo([string] $pathToRepo)
+{
+    $repoInfoFile = Combine $pathToRepo "repoInfo"
+
+    if (Test-Path $repoInfoFile)
+    {
+        $repoInfo = [PSCustomObject](Get-Content $repoInfoFile | ConvertFrom-Json)
+
+        $repoDirectory = Combine $pathToRepo "repo"
+        $repoInfo | Add-Member -MemberType NoteProperty -Name "RepoDirectory" -Value $repoDirectory
+
+        if ($null -ne $repoInfo.SolutionFile)
+        {
+            $repoInfo.SolutionFile = Combine $repoInfo.RepoDirectory $repoInfo.SolutionFile
+        }
+
+        return $repoInfo
+    }
+    else
+    {
+        return [PSCustomObject]@{
+            RepoAddress = $null
+            RepoLocation = $null
+            SolutionFile = $null
+            RepoDirectory = $null
+        }
+    }
+}
+
+function MaterializeRepo([PSCustomObject] $repoInfo, [string] $repoRoot)
+{
+    $repoAddress = $repoInfo.RepoAddress
+    $repoCommit = $repoInfo.RepoLocation
+    $repoDirectory = $repoInfo.RepoDirectory
+
+    CloneOrUpdateRepo $repoAddress $repoCommit $repoDirectory
+}
+
+function MaterializeRepoIfNecessary([PSCustomObject]$repoInfo)
+{
+    if ($null -ne $repoInfo.RepoAddress)
+    {
+        Write-Information "Materializing $($repoInfo.RepoAddress)"
+        MaterializeRepo $repoInfo $projectRoot
```

Comment:
This looks odd, where is `$projectRoot` defined? How does this work? :) 
Can you add it as an explicit parameter and explicitly pass it from callsites?

Reply:
The wonderful world of powershell. Not intended, it just happened to work.
The funny part is that I discovered the same type of problem on the existing Test.ps1, which I fixed.